core/src/main/scala/cats/data/AndThen.scala

```diff
@@ -0,0 +1,120 @@
+package cats.data
+
+import java.io.Serializable
+
+/**
+ * Internal API (Cats) — A type-aligned seq for representing
```

Comment:
This actually seems like something to move to kernel since it is not higher kinded very useful.

I see it is private, but lots of use cases could exist for an optimized AndThen.

Reply:
It is not very clear to me what the boundaries of Cats's sub-projects and packages are. 

I can move it to cats-kernel if needed, however I only see a bunch of type classes in cats-kernel, whereas I am seeing data types that aren't type classes in `cats.data`.